packages/server/src/workers/cron.ts

```diff
@@ -0,0 +1,223 @@
+import { createReference } from '@medplum/core';
+import { Bot, Resource, Timing } from '@medplum/fhirtypes';
+import { Job, Queue, QueueBaseOptions, Worker } from 'bullmq';
+import { MedplumRedisConfig } from '../config';
+import { executeBot, isBotEnabled } from '../fhir/operations/execute';
+import { systemRepo } from '../fhir/repo';
+import { logger } from '../logger';
+import { AuditEventOutcome } from '../util/auditevent';
+import { isValidCron } from 'cron-validator';
+import { createAuditEvent, findProjectMembership } from './utils';
+
+const daysOfWeekConversion = { sun: 0, mon: 1, tue: 2, wed: 3, thu: 4, fri: 5, sat: 6 };
+
+/*
+ * The Cron worker inspects resources takes a bot,
+ * if it has the Cron property, will add it as a repeatable
+ * Cron job
+ */
+// Repeatable is based on BullMQ docs https://docs.bullmq.io/guide/jobs/repeatable
+interface Repeatable {
+  repeat: {
+    pattern?: string;
+    every?: number;
+    limit?: number;
+  };
+  jobId?: string;
+}
+
+export interface CronJobData {
+  readonly resourceType: string;
+  readonly id: string;
+}
+
+const queueName = 'CronQueue';
+const jobName = 'CronJobData';
+
+let queue: Queue<CronJobData> | undefined = undefined;
+let worker: Worker<CronJobData> | undefined = undefined;
+
+/**
+ * Initializes the Cron worker.
+ * Sets up the BullMQ job queue.
+ * Sets up the BullMQ worker.
+ */
+export function initCronWorker(config: MedplumRedisConfig): void {
+  const defaultOptions: QueueBaseOptions = {
+    connection: config,
+  };
+
+  queue = new Queue<CronJobData>(queueName, {
+    ...defaultOptions,
+    defaultJobOptions: {
+      attempts: 3,
+      backoff: {
+        type: 'exponential',
+        delay: 1000,
+      },
+    },
+  });
+  worker = new Worker<CronJobData>(queueName, execBot, defaultOptions);
+  worker.on('completed', (job) => logger.info(`Completed job ${job.id} successfully`));
+  worker.on('failed', (job, err) => logger.info(`Failed job ${job?.id} with ${err}`));
+}
+
+/**
+ * Shuts down the Cron worker.
+ * Closes the BullMQ job queue.
+ * Clsoes the BullMQ worker.
+ */
+export async function closeCronWorker(): Promise<void> {
+  if (queue) {
+    await queue.close();
+    queue = undefined;
+  }
+
+  if (worker) {
+    await worker.close();
+    worker = undefined;
+  }
+}
+
+/**
+ * Returns theCron queue instance.
+ * This is used by the unit tests.
+ * @returns The Cron queue (if available).
+ */
+export function getCronQueue(): Queue<CronJobData> | undefined {
+  return queue;
+}
+
+/**
+ * @param resource The resource that was created or updated.
+ */
+export async function addCronJobs(resource: Resource): Promise<void> {
+  if (resource.resourceType !== 'Bot') {
+    // For now we have only the bot to execute on a timed job
+    return;
+  }
+  const bot = resource;
+  const botEnabled = await isBotEnabled(bot);
```

Comment:
Rather than using `isBotEnabled`, let's do a different feature flag check.  Something like:

```ts
  const project = await systemRepo.readResource<Project>('Project', resource.meta?.project as string);
  const isCronEnabled = !!project.features?.includes('cron');
```